packages/coding-standards/src/Sniffs/ForbiddenDoctrineInheritanceSniff.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\CodingStandards\Sniffs;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class ForbiddenDoctrineInheritanceSniff implements Sniff
+{
+    /**
+     * @return int[]
+     */
+    public function register()
+    {
+        return [T_CLASS];
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $file
+     * @param int $classPosition
+     */
+    public function process(File $file, $classPosition)
+    {
+        $phpDocStartPosition = $file->findPrevious(T_DOC_COMMENT_OPEN_TAG, $classPosition);
+        if ($phpDocStartPosition === false) {
+            return;
+        }
+
+        $phpDocTags = $this->findPhpDocTags($file, $classPosition, $phpDocStartPosition);
+        foreach ($phpDocTags as $position => $token) {
+            if ($this->isDoctrineInheritanceAnnotation($token)) {
+                $message = 'It is forbidden to use Doctrine inheritance mapping because it causes problems during entity extension. Such problem with `OrderItem` was resolved during making OrderItem extendable #715.';
```

Comment:
Do we really need to explain why have this sniffer?

Reply:
yep. Check our internal system - notes https://app.yodiz.com/plan/pages/board.vz?cid=10273#/app/us-5241